src/main/java/com/jnape/palatable/shoki/api/MultiSet.java

```diff
@@ -107,4 +112,137 @@
+     * or <code>ys</code> with the occurrences of whichever one had more elements. This is equivalent to
+     * {@link MultiSet#merge} partially applied with {@link com.jnape.palatable.lambda.semigroup.builtin.Max#max}.
+     *
+     * @param other the {@link MultiSet} to union with this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> union(MultiSet<A> other) {
+        return merge(max(), other);
+    }
+
+    /**
+     * The <a href="https://en.wikipedia.org/wiki/Complement_(set_theory)#Relative_complement"
+     * target="_new">difference</a> (also known as "relative complement") of two {@link MultiSet MultiSets}
+     * <code>xs</code> and <code>ys</code> is the {@link MultiSet} of in <code>xs</code> with all elements of
+     * <code>ys</code> removed.
+     *
+     * @param other the {@link MultiSet} to subtract from this {@link MultiSet}
+     * @return the difference {@link MultiSet}
+     */
+    default MultiSet<A> difference(MultiSet<A> other) {
+        return foldLeft(curried(ms -> into(ms::remove)), this, other);
+    }
+
+    /**
+     * Determine if another {@link MultiSet} is a subset of this {@link MultiSet}. A {@link MultiSet} <code>x</code> is
+     * a subset of <code>y</code> if for every <code>a^i</code> in <code>x</code> there exists <code>a^j</code> in
+     * <code>y</code> where ^ denotes the number of occurrences in the multiset.
+     *
+     * @param other the {@link MultiSet} to test whether it's a subset
+     * @return whether other is a subset
+     */
+    default boolean inclusion(MultiSet<A> other) {
+        return and().foldMap(into(this::contains), other);
+    }
+
+    /**
+     * The union of two {@link MultiSet MultiSets}
+     * <code>xs</code> and <code>ys</code> is the {@link MultiSet} of those elements present in both <code>xs</code>
+     * and <code>ys</code> with the occurrences of whichever one had less elements. This is equivalent to
+     * {@link MultiSet#merge} partially applied with {@link com.jnape.palatable.lambda.semigroup.builtin.Min#min}.
+     *
+     * @param other the {@link MultiSet} to intersect with this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> intersection(MultiSet<A> other) {
+        return merge(min(), other);
+    }
+
+
+    /**
+     * The symmetric difference of two {@link MultiSet MultiSets} <code>xs</code> and <code>ys</code> is the
+     * {@link MultiSet} of the occurrence elements present in either <code>xs</code> or <code>ys</code> but not the
+     * other.
+     *
+     * @param other the {@link MultiSet} to symmetrically subtract from this {@link MultiSet}
+     * @return the union {@link MultiSet}
+     * @see MultiSet#merge
+     */
+    default MultiSet<A> symmetricDifference(MultiSet<A> other) {
+        return difference(other).union(other.difference(this));
```

Comment:
This stands out to me as another case where the `union` definition for `MultiSet` being the supremum makes sense, and the symmetry with `Set` is intuitive. Nice.